src/plugins/teleop/Teleop_TEST.cc

```diff
@@ -0,0 +1,442 @@
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      QCoreApplication::processEvents();
+      sleep++;
+    }
+
+    EXPECT_TRUE(received);
+    received = false;
+
+    // Generates a key release event on the main window.
+    QKeyEvent *keyrelease_A = new QKeyEvent(QKeyEvent::KeyRelease,
+      Qt::Key_A, Qt::NoModifier);
+    app.sendEvent(win->QuickWindow(), keyrelease_A);
+
+    sleep = 0;
+    // cppcheck-suppress knownConditionTrueFalse
+    while (!received && sleep < maxSleep)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      QCoreApplication::processEvents();
+      sleep++;
+    }
+
+    EXPECT_TRUE(received);
+    received = false;
+    // Generates a key release event on the main window.
+    QKeyEvent *keyrelease_W = new QKeyEvent(QKeyEvent::KeyRelease,
+      Qt::Key_W, Qt::NoModifier);
+    app.sendEvent(win->QuickWindow(), keyrelease_W);
+
+    sleep = 0;
+    // cppcheck-suppress knownConditionTrueFalse
+    while (!received && sleep < maxSleep)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      QCoreApplication::processEvents();
+      sleep++;
+    }
+
+    EXPECT_TRUE(received);
+    received = false;
+    // Generates a key press event on the main window.
+    QKeyEvent *keypress_X = new QKeyEvent(QKeyEvent::KeyPress,
+      Qt::Key_X, Qt::NoModifier);
+    app.sendEvent(win->QuickWindow(), keypress_X);
+
+    sleep = 0;
+    // cppcheck-suppress knownConditionTrueFalse
+    while (!received && sleep < maxSleep)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      QCoreApplication::processEvents();
+      sleep++;
+    }
+
+    EXPECT_TRUE(received);
+    received = false;
+    // Generates a key release event on the main window.
+    QKeyEvent *keyrelease_X = new QKeyEvent(QKeyEvent::KeyRelease,
+      Qt::Key_X, Qt::NoModifier);
+    app.sendEvent(win->QuickWindow(), keyrelease_X);
+
+    sleep = 0;
+    // cppcheck-suppress knownConditionTrueFalse
+    while (!received && sleep < maxSleep)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      QCoreApplication::processEvents();
+      sleep++;
+    }
+
+    // Cleanup
+    plugins.clear();
+  };
+
+};
+
+/////////////////////////////////////////////////
+TEST_F(TeleopTest, Load)
+{
+  Application app(g_argc, g_argv);
+  app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+  EXPECT_TRUE(app.LoadPlugin("Teleop"));
+
+  // Get main window
+  auto win = app.findChild<MainWindow *>();
+  ASSERT_NE(nullptr, win);
+
+  // Get plugin
+  auto plugins = win->findChildren<Plugin *>();
+  EXPECT_EQ(plugins.size(), 1);
+
+  auto plugin = plugins[0];
+  EXPECT_EQ(plugin->Title(), "Teleop");
+
+  // Cleanup
+  plugins.clear();
+}
+
+/////////////////////////////////////////////////
+TEST_F(TeleopTest, ButtonCommand)
+{
+  TestButtons("/cmd_vel");
+  TestButtons("/model/vehicle_blue/cmd_vel");
```

Comment:
Running it only once is enough I consider.

Reply:
Addressed